ejb3/src/main/java/org/jboss/as/ejb3/tx/CMTTxInterceptor.java

```diff
@@ -83,15 +83,16 @@ protected void endTransaction(final TransactionManager tm, final Transaction tx)
                 throw EjbLogger.ROOT_LOGGER.wrongTxOnThread(tx, tm.getTransaction());
             }
             final int txStatus = tx.getStatus();
-            if (txStatus == Status.STATUS_ACTIVE) {
+            if (txStatus == Status.STATUS_ACTIVE || txStatus == Status.STATUS_PREPARED || txStatus == Status.STATUS_COMMITTED) {
```

Comment:
done - please do review it though because I can only work on the status' and would appreciate your input to verify it is doing what I expect.

I would say the preparing/prepared/committing/committed should never happen in CMT (unless some as yet undeveloped feature in narayana that somehow triggered an auto-commit was enabled).

I initially thought that is what you were commenting on - that a transaction which was already committed in CMT should throw an exception. I could be persuaded that would be the right thing to do.

ROLLING_BACK needs the rollback call to get the InvalidStateException.